cvat-ui/src/utils/hooks.ts

```diff
@@ -130,3 +132,23 @@ export function useAuthQuery(): Record<string, string> | null {
     const queryParams = new URLSearchParams(history.location.search);
     return authQuery(queryParams);
 }
+
+export function useDynamicLabels(componentShortcuts: Record<string, KeyMapItem>): void {
+    const keyMap = useSelector((state: CombinedState) => state.shortcuts.keyMap);
+    const keyMapRef = useRef(keyMap);
+
+    useEffect(() => {
+        keyMapRef.current = keyMap;
+    }, [keyMap]);
+
+    useEffect(() => () => {
+        const revertedShortcuts = Object.entries(componentShortcuts).reduce((acc: any, [key, value]) => {
+            acc[key] = {
+                ...value,
+                sequences: keyMapRef.current[key] ? keyMapRef.current[key].sequences : value.sequences,
```

Comment:
```suggestion
                sequences: keyMapRef.current[key]?.sequences ?? value.sequences,
```

Reply:
Also I think that taking `value.sequences` as default value may be incorrect choice as it can lead to conflicts with registered shortcuts. Thus, better to take empty array, don't you think?

Reply:
I'll check